### PR TITLE
Issue #SB-24563 fix: Getting Failed error while trying to publish questionSet

### PR DESCRIPTION
**Pull request by rajnishdargan**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-24563" title="SB-24563" target="_blank">BugSB-24563</a>  VDN - Getting Failed error while trying to publish questionSet
  </td></table>
  <br />
 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions: Angular 8, Nodejs 12.16.1
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/client/src/app/modules/sourcing/components/mcq-creation/mcq-creation.component.ts [hunk 1]
```diff
@@ -277,6 +277,19 @@ export class McqCreationComponent implements OnInit, OnChanges, AfterViewInit {
       }
     }
   }
+
+  validateCurrentQuestion() {
+    const optionValid = _.find(this.mcqForm.options, option =>
+      (option.body === undefined || option.body === '' || option.length > this.setCharacterLimit));
+    if (optionValid || !this.mcqForm.answer || [undefined, ''].includes(this.mcqForm.question)) {
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/components/mcq-creation/mcq-creation.component.ts:284]: Change the variable name to `optionInvalid` if you are looking for falsy values.

File: src/app/client/src/app/modules/sourcing/components/question-creation/question-creation.component.ts [hunk 2]
```diff
@@ -332,6 +332,16 @@ export class QuestionCreationComponent implements OnInit, AfterViewInit, OnChang
     }
   }
 
+  validateCurrentQuestion() {
+    if (this.editorState.question === '' || this.editorState.answer === '') {
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/components/question-creation/question-creation.component.ts:336]: can we use `_.isEmpty` just for the sake of defensive programming.

**Review by venkateshwarans — COMMENTED** [review]
check comments

File: src/app/client/src/app/modules/sourcing/components/mcq-creation/mcq-creation.component.ts (hunk 1 above)
Comment (rajnishdargan) [reply src/app/client/src/app/modules/sourcing/components/mcq-creation/mcq-creation.component.ts]: done

File: src/app/client/src/app/modules/sourcing/components/question-creation/question-creation.component.ts (hunk 2 above)
Comment (rajnishdargan) [reply src/app/client/src/app/modules/sourcing/components/question-creation/question-creation.component.ts]: done

**Review by pallakartheekreddy — APPROVED** [review]

**Pull request merged**